added example how to add username and password to a request

gertminov opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/fastify/fastify/blob/master/CONTRIBUTING.md

By making a contribution to this project, I certify that:

* (a) The contribution was created in whole or in part by me and I
  have the right to submit it under the open source license
  indicated in the file; or

* (b) The contribution is based upon previous work that, to the best
  of my knowledge, is covered under an appropriate open source
  license and I have the right under that license to submit that
  work with modifications, whether created in whole or in part
  by me, under the same open source license (unless I am
  permitted to submit under a different license), as indicated
  in the file; or

* (c) The contribution was provided directly to me by some other
  person who certified (a), (b) or (c) and I have not modified
  it.

* (d) I understand and agree that this project and the contribution
  are public and that a record of the contribution (including all
  personal information I submit with it, including my sign-off) is
  maintained indefinitely and may be redistributed consistent with
  this project or the open source license(s) involved.
-->

resolves #78 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

gertminov committed:
added example how to add username and password to a request

mcollina reviewed (approved):
lgtm

jsumners commented on README.md:
```diff
@@ -99,6 +99,30 @@ fastify.after(() => {
+### How to add credentials to a request
+The basic-auth plugin retrieves the credentials from the [Authorization](https://en.wikipedia.org/wiki/Basic_access_authentication) header.
+
+A simple example to add `username` and `password` to the reply header:
```
Why would we give an example showing sending the credentials in a response header? https://www.rfc-editor.org/rfc/rfc7617.html does not call for this, and it seems like a bad practice to recommend.

jsumners reviewed (commented):
I am unclear about the response example, and I think the plugin is pretty clear that it implements basic authentication (i.e. https://www.rfc-editor.org/rfc/rfc7617.html). So it seems redundant to me to add an example, but I suppose it doesn't hurt anything either. I would just remove the "send credentials back the client" portion as it isn't spec compliant and is a glaring security violation.

Eomm commented on README.md:
````diff
@@ -99,6 +99,30 @@ fastify.after(() => {
+
+An example to add `username` and `password` via [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch):
+```typescript
+fetch('URL_GOES_HERE', { 
````
```suggestion
fetch('https://www.example.org/', { 
```

gertminov committed:
applied suggestions

- removed node example
- changed misspelling
- added spacing
- changed 'URL_GOES_HERE' to 'https://www.example.org/'
